src/drivers/mediatek/afe/afe-drv.c

```diff
@@ -54,29 +55,31 @@ static int afe_memif_set_channel(struct mtk_base_afe *afe, int id, unsigned int
 	struct mtk_base_afe_memif *memif = &afe->memif[id];
 	unsigned int mono;
 
-	if (memif->data->mono_shift < 0)
-		return 0;
-
 	if (memif->data->ch_num_reg >= 0) {
 		afe_reg_update_bits(afe, memif->data->ch_num_reg,
 				    memif->data->ch_num_maskbit << memif->data->ch_num_shift,
 				    channel << memif->data->ch_num_shift);
 	}
 
 	if (memif->data->quad_ch_mask) {
-		unsigned int quad_ch = 0;
-
-		if (channel == 4)
-			quad_ch = 1;
+		unsigned int quad_ch = (channel == 4) ? 1 : 0;
```

Comment:
all these differences are because you're merging two versions or you're moving and changing?
BTW, this line could just be `quad_ch = channel == 4;`